test/VoucherHub.test.ts

```diff
@@ -787,6 +789,67 @@ describe('VoucherHub', function () {
+        let voucherHub: VoucherHub;
+
+        beforeEach(async function () {
+            ({ voucherHub, voucherOwner1, anyone } = await loadFixture(deployFixture));
+            // Create voucher type
+            await voucherHubWithAssetEligibilityManagerSigner
+                .createVoucherType(description, duration)
+                .then((tx) => tx.wait());
+            // Add eligible asset
+            await voucherHubWithAssetEligibilityManagerSigner
+                .addEligibleAsset(voucherType, asset)
+                .then((tx) => tx.wait());
+            // Create voucher
+            voucher = await voucherHubWithVoucherManagerSigner
+                .createVoucher(voucherOwner1, voucherType, voucherValue)
+                .then((tx) => tx.wait())
+                .then(() => voucherHub.getVoucher(voucherOwner1))
+                .then((voucherAddress) => ethers.getImpersonatedSigner(voucherAddress));
+        });
+
+        it('Should refund voucher', async function () {
+            const debitedValue = BigInt(assetPrice * 3);
+            const voucherInitialCreditBalance = await voucherHub.balanceOf(voucher.address);
+            await voucherHub
+                .connect(voucher)
+                .debitVoucher(voucherType, asset, assetPrice, asset, assetPrice, asset, assetPrice)
+                .then((tx) => tx.wait());
+            expect(await voucherHub.balanceOf(voucher.address)).equals(
+                voucherInitialCreditBalance - debitedValue,
+            );
+            // Refund voucher.
+            const refundAmount = debitedValue / 2n; // any amount < sponsoredValue
+            await expect(voucherHub.connect(voucher).refundVoucher(refundAmount))
+                .to.emit(voucherHub, 'Transfer')
+                .withArgs(ethers.ZeroAddress, voucher.address, refundAmount)
+                .to.emit(voucherHub, 'VoucherRefunded')
+                .withArgs(voucher.address, refundAmount);
+        });
+
+        it('Should not refund when sender is not a voucher', async function () {
+            const debitedValue = BigInt(assetPrice * 3);
+            const voucherInitialCreditBalance = await voucherHub.balanceOf(voucher.address);
+            await voucherHub
+                .connect(voucher)
+                .debitVoucher(voucherType, asset, assetPrice, asset, assetPrice, asset, assetPrice)
+                .then((tx) => tx.wait());
+            expect(await voucherHub.balanceOf(voucher.address)).equals(
+                voucherInitialCreditBalance - debitedValue,
+            );
+            // Refund voucher.
+            const refundAmount = debitedValue / 2n; // any amount < sponsoredValue
+            await expect(voucherHub.connect(anyone).refundVoucher(refundAmount)).to.be.revertedWith(
+                'VoucherHub: sender is not voucher',
+            );
+        });
+
+        it.skip('TODO ?? - Should not refund voucher when amount is greater than voucher value', async function () {});
```

Comment:
To be removed?

Reply:
Yes :+1: [Clean](https://github.com/iExecBlockchainComputing/iexec-voucher-contracts/pull/20/commits/cd2e63ff841ca66360b2e9bea1713f13f0a14813)